src/relative_duration/parse.rs

```diff
@@ -0,0 +1,191 @@
+    /// [ISO 8601 duration string](https://en.wikipedia.org/wiki/ISO_8601#Durations).
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use chronoutil::RelativeDuration;
+    ///
+    /// assert_eq!(
+    ///     RelativeDuration::years(1).to_iso_8601(),
+    ///     "P1Y",
+    /// );
+    /// ```
+    pub fn to_iso_8601(&self) -> String {
+        let years = self.months as i64 / 12;
+        let months = self.months as i64 % 12;
+
+        let duration_seconds = self.duration.num_seconds();
+
+        // TODO: address this unwrap
+        let days = duration_seconds / (24 * 60 * 60);
+        let mut remaining_seconds = duration_seconds - (days * 24 * 60 * 60);
+
+        let hours = remaining_seconds / (60 * 60);
+        remaining_seconds -= hours * 60 * 60;
+
+        let minutes = remaining_seconds / 60;
+        remaining_seconds -= minutes * 60;
+
+        let seconds = remaining_seconds;
+
+        let date_spec = format_spec([years, months, days], ['Y', 'M', 'D']);
+        let time_spec = format_spec([hours, minutes, seconds], ['H', 'M', 'S']);
+
+        let mut out = String::with_capacity(date_spec.len() + time_spec.len() + 2);
+        out.push('P');
+        out.push_str(&date_spec);
+        if !time_spec.is_empty() {
+            out.push('T');
+            out.push_str(&time_spec);
+        }
+
+        out
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_parse_duration() {
+        [
+            (
+                "P1YT1S",
+                RelativeDuration::months(12).with_duration(Duration::seconds(1)),
+            ),
+            (
+                "P2Y2M2DT2H2M2S",
+                RelativeDuration::months(2 * 12 + 2).with_duration(dhms_to_duration(2, 2, 2, 2)),
+            ),
+            (
+                "P1M",
+                RelativeDuration::months(1).with_duration(Duration::zero()),
+            ),
+            ("PT10M", RelativeDuration::minutes(10)),
+            ("P-1M", RelativeDuration::months(-1)),
+            ("P1W1D", RelativeDuration::days(8)),
+        ]
+        .iter()
+        .for_each(|(input, expected)| {
+            assert_eq!(RelativeDuration::from_iso_8601(input).unwrap(), *expected)
+        })
+    }
+
+    #[test]
+    fn test_format_duration() {
+        [
+            (
+                RelativeDuration::months(12).with_duration(Duration::seconds(1)),
+                "P1YT1S",
+            ),
+            (
+                RelativeDuration::months(2 * 12 + 2).with_duration(dhms_to_duration(2, 2, 2, 2)),
+                "P2Y2M2DT2H2M2S",
+            ),
+            (
+                RelativeDuration::months(1).with_duration(Duration::zero()),
+                "P1M",
+            ),
+            (RelativeDuration::minutes(10), "PT10M"),
+            (RelativeDuration::months(-1), "P-1M"),
+        ]
+        .iter()
+        .for_each(|(input, expected)| assert_eq!(input.to_iso_8601(), *expected))
+    }
+}
```

Comment:
A few more test cases would be good. In particular some of the cancellation logic between the various fields in an input like `P1Y-10M-1W3DT3H-6M-1S` and a longer negative duration like `P-23M`.

Reply:
Done.